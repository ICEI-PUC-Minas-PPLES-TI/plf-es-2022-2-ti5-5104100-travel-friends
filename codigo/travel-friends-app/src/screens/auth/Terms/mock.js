export const term =
  'Os serviços do Travel Friends são fornecidos pela pessoa jurídica com a seguinte razão social:Travel Friends LTDA, como o nome fantasia Travel Friends, inscrito no CNPJ sob o nº 47.859.546/0001-66, titular da propriedade intelectual sobre software, website, aplicativos, conteúdos e demais ativos relacionados à plataforma Travel Friends.';
