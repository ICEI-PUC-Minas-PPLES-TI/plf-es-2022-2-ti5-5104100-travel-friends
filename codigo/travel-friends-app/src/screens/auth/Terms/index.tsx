import React from "react";
import { View, Text } from "react-native";
import { useInitialNavigation } from "../../../hooks/navigation";
import { term } from "./mock";

import Button from "../../../components/Button";

import { styles } from "./styles";

export default function Terms() {
  const navigation = useInitialNavigation();

  return (
    <View style={styles.container}>
      <View style={styles.header}>
        <Text style={styles.title}>Termo de uso</Text>
      </View>
      <View style={styles.content}>
        <Text style={styles.textContent}>{term}</Text>
        <Button title="Aceitar" onPress={() => navigation.navigate("SignIn")} />
      </View>
    </View>
  );
}
