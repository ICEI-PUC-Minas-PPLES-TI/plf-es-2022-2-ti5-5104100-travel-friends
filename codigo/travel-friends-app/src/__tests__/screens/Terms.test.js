import React from "react";
import renderer from "react-test-renderer";

import Terms from "../../screens/auth/Terms";

describe("<Terms />", () => {
  it("has 1 child", () => {
    const tree = renderer.create(<Terms />).toJSON();
    expect(tree.children.length).toBe(3);
  });
  it("renders correctly", () => {
    const tree = renderer.create(<Terms />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
