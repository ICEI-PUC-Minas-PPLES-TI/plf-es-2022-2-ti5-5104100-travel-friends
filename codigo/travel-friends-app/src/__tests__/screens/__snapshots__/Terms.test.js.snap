// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Terms /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#eeeefd",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "padding": 20,
        "paddingVertical": 60,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#101d42",
          "fontSize": 34,
          "fontWeight": "600",
        }
      }
    >
      Termo de uso
    </Text>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#fafcfd",
            "borderTopLeftRadius": 16,
            "borderTopRightRadius": 16,
            "flex": 1,
            "justifyContent": "space-around",
            "padding": 20,
            "paddingTop": 40,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#818191",
              "textAlign": "auto",
            }
          }
        >
          Os serviços do Travel Friends são fornecidos pela pessoa jurídica com a seguinte razão social:Travel Friends LTDA, como o nome fantasia Travel Friends, inscrito no CNPJ sob o nº 47.859.546/0001-66, titular da propriedade intelectual sobre software, website, aplicativos, conteúdos e demais ativos relacionados à plataforma Travel Friends. 1. Do objeto: a plataforma visa licenciar o uso de seu software, website, aplicativos e demais ativos de propriedade intelectual, fornecendo ferramentas para auxiliar e dinamizar o dia a dia dos seus usuários. A plataforma caracteriza-se pela prestação do seguinte serviço: conexão de pessoas e lugares. A plataforma não realiza a distância por meio eletrônico de produtos ou serviços. 2. Da aceitação: o presente termo estabelece obrigações contratadas de livre e espontânea vontade, por tempo indeterminado, entre a plataforma e as pessoas físicas ou jurídicas, usuárias do OU site OU aplicativo. Ao utilizar a plataforma o usuário aceita integralmente as presentes normas e compromete-se a observá-las, sob o risco de aplicação das penalidades cabíveis. A aceitação do presente instrumento é imprescindível para o acesso e para a utilização de quaisquer serviços fornecidos pela empresa. Caso não concorde com as disposições deste instrumento, o usuário não deve utilizá-los. 3. Do acesso dos usuários: serão utilizadas todas as soluções técnicas à disposição do responsável pela plataforma para permitir o acesso ao serviço 24 (vinte e quatro) horas por dia, 7(sete) dias por semana. No entanto, a navegação na plataforma ou em alguma de suas páginas poderá ser interrompida, limitada ou suspensa para atualizações, modificações ou qualquer ação necessária ao seu bom funcionamento.
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "#fafcfd",
        "paddingVertical": 30,
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#3f48cc",
          "borderRadius": 8,
          "justifyContent": "center",
          "opacity": 1,
          "paddingVertical": 10,
          "width": 660,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 22,
          }
        }
      >
        Aceitar
      </Text>
    </View>
  </View>
</View>
`;
